### PR TITLE
Moves crud association templates to admin-bundle

### PR DESCRIPTION
**Pull request by ju1ius**
This PR is related to [SonataAdminBundle#3971](https://github.com/sonata-project/SonataAdminBundle/pull/3971)
I am targetting this branch, because there are BC breaks
## Changelog

``` markdown
### Added
- `FieldDescription` implements `FieldDescriptionInterface::describesSingleValuedAssociation` and `FieldDescriptionInterface::describesCollectionValuedAssociation` methods

### Changed
- `FormContractor` now extends `Sonata\AdminBundle\Admin\AbstractFormContractor`

### Removed
- removed ALL crud association templates and form widget blocks. They now belong to the admin bundle
```

### CODE REVIEW

**Comment by OskarStark** [comment]
@ju1ius i removed the `### Deprecated` section from your PR header, because it was empty 👍

**Comment by dbu** [comment]
deleting tons of stuff, loving it!

can you adjust composer.json to temporarily use your pull request against the admin bundle? then we see if it already works out.
once that is merged, you can update the composer.json again to require the next version of admin bundle that contains the consolidated files.

do you have a demo application to validate if things indeed still work? particularly the js interaction is not covered by automated tests...

**Comment by ju1ius** [comment]
> can you adjust composer.json to temporarily use your pull request against the admin bundle?

@dbu  Sure, will do, please wait a few minutes

> do you have a demo application to validate if things indeed still work?

You mean something for you to download ?

**Comment by dbu** [comment]
> You mean something for you to download ?

no, more to know if you can test what you do on phpcr admin side.
i will for sure try this with the symfony cmf sandbox when you declare
it ready.

**Comment by ju1ius** [comment]
> to know if you can test what you do on phpcr admin side

@dbu Yes, I developed these 4 PRs inside a test project's vendor directory.
I initially tried to do this inside the cmf-sandbox, but being a total phpcr noob, I found it to be too complex a starting point. So I installed just the minimal set of bundles to get on tracks quickly.

Since I'm not familiar with PHPCR, the model I use is basically MongoDB documents with the mapping adapted for PHPCR. Enough to allow me to test the widgets, but not very content-repository-ish I guess...

That being said, I'm definitely waiting for your input on this one. :wink: 
For example I based the [new FieldDescription methods](https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/395/files#diff-342d4c5b8800a668d40523101d2cd5ccR120) on what you told me in #389, which I believe to be conceptually correct, but I'm still not 100% sure it is accurate in _this_ particular context. Like do we actually need to handle the `child`, `parent`, `mixedreferrers` mapping types ? Does it make sense to use anything else than `doctrine_phpcr_odm_tree` for a `parent` field ?

**Comment by ju1ius** [comment]
> can update the composer.json again to require the next version of admin bundle

Shall I use this alias
```json
"sonata-project/admin-bundle": "dev-refactor/bring-templates-back as 3.x-dev"
```
or this one:
```json
"sonata-project/admin-bundle": "dev-refactor/bring-templates-back as 3.1"
```

**Comment by dbu** [comment]
> "sonata-project/admin-bundle": "dev-refactor/bring-templates-back as 3.1"

looks good, if you don't end up with composer conflicts. basically use
whatever is needed to make composer install, for now :-)

File: Admin/FieldDescription.php [hunk 1]
```diff
@@ -115,4 +116,29 @@ public function setParentAssociationMappings(array $parentAssociationMappings)
...
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function describesSingleValuedAssociation()
+    {
+        return in_array($this->mappingType, array(
+            ClassMetadata::MANY_TO_ONE,
+            'child',
```
Comment (dbu) [inline Admin/FieldDescription.php]: actually not sure if this is not ONE_TO_ONE. a child can only have one parent, and a parent can only have a particular child once.

**Comment by dbu** [comment]
i think the tree is the right default widget for parent. i assume its still possible to specify a widget with sonata if i need something different in some situation? (e.g. for the parent, i might have a limited list of options that i want to show as dropdown)

**Comment by dbu** [comment]
@WouterJ did you see this? do you have input?

**Comment by ju1ius** [comment]
> can you adjust composer.json

@dbu done!

**Comment by ju1ius** [comment]
@dbu @Soullivaneuh I updated the composer.json to require `ju1ius/SonataAdminBundle@dev-refactor/bring-templates-back`, but there are still [a few builds](https://travis-ci.org/sonata-project/SonataDoctrinePhpcrAdminBundle/builds/139974426) failing (because they refuse to load my branch I guess).
Any idea on how to fix that ?

**Comment by Soullivaneuh** [comment]
@ju1ius the build with `SONATA_ADMIN=3.*` target failed because the `admin-bundle` dependency is replaced.

Simply ignore it for now. It should be OK when you will fallback to the classic dependency.

**Comment by core23** [comment]
@ju1ius Can you give us a status update on this? 

You did great work so far, please continue working on this 👍 👍 👍

**Comment by greg0ire** [comment]
@core23 I think we should not kid ourselves and close this, but maybe someone would like to carry this PR?

**Pull request closed without merging**